pkg/apply/solver/solver.go

```diff
@@ -209,9 +210,13 @@ func (t *TaskQueueBuilder) AppendPruneTask(pruneObjs object.UnstructuredSet,
 // (like CRD's). Returns a pointer to the Builder to chain function calls.
 func (t *TaskQueueBuilder) AppendApplyWaitTasks(applyObjs object.UnstructuredSet,
 	applyFilters []filter.ValidationFilter, applyMutators []mutator.Interface, o Options) *TaskQueueBuilder {
-	// Use the "depends-on" annotation to create a graph, ands sort the
-	// objects to apply into sets using a topological sort.
-	applySets, err := graph.SortObjs(applyObjs)
+	// Add dependencies from annotations
```

Comment:
AddDependencies moved to the Applier/Destroyer. 
I don't like it there, but I'm not sure where a better place to put it is that has both `applyObjs` and `pruneObjs`.